fenix/app/src/main/java/org/mozilla/fenix/ext/NavController.kt

```diff
@@ -36,6 +39,29 @@ fun NavController.navigateSafe(
     }
 }
 
+/**
+ * Navigates using the given [directions], and submit a Breadcrumb
+ * when an [IllegalArgumentException] happens.
+ */
+fun NavController.navigateWithBreadcrumb(
+    directions: NavDirections,
+    navigateFrom: String,
+    navigateTo: String,
+    crashReporter: CrashReporter,
+) {
+    try {
+        this.navigate(directions)
+    } catch (e: IllegalArgumentException) {
```

Comment:
Ack, @Amejia481 I realized that this is a catch so the crash exception won't end up in Sentry. We probably need to add the breadcrumb, and then let the crash happen.

Reply:
Yeah you are right, I think we missed the submit the exception.